node/libs/roles/src/validator/messages/consensus.rs

```diff
@@ -74,25 +76,48 @@ impl Default for Fork {
 /// A struct that represents a set of validators. It is used to store the current validator set.
 /// We represent each validator by its validator public key.
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub struct ValidatorSet {
+pub struct ValidatorCommittee {
     vec: Vec<validator::PublicKey>,
-    map: BTreeMap<validator::PublicKey, usize>,
+    indexes: BTreeMap<validator::PublicKey, usize>,
+    weights: Vec<usize>,
 }
 
-impl ValidatorSet {
-    /// Creates a new ValidatorSet from a list of validator public keys.
-    pub fn new(validators: impl IntoIterator<Item = validator::PublicKey>) -> anyhow::Result<Self> {
+impl ValidatorCommittee {
+    /// Maximum validator weight
+    /// 100.00%
+    pub const MAX_WEIGHT: usize = 10000;
+
+    /// Required weight to verify signatures.
+    /// Currently 80.00%
+    const THRESHOLD: usize = 8000;
+
+    /// Creates a new ValidatorCommittee from a list of validator public keys.
+    pub fn new(validators: impl IntoIterator<Item = WeightedValidator>) -> anyhow::Result<Self> {
         let mut set = BTreeSet::new();
+        let mut weights = BTreeMap::new();
         for validator in validators {
-            anyhow::ensure!(set.insert(validator), "Duplicate validator in ValidatorSet");
+            anyhow::ensure!(
+                set.insert(validator.key.clone()),
+                "Duplicate validator in ValidatorCommittee"
+            );
+            weights.insert(
+                validator.key,
```

Comment:
you should check that the weights are from the allowed range and that the sum is 1. Also add a test for those checks

Reply:
Disagree. There's no need to make the weights add up to 1.